src/app/core/internal/impl/CircuitInternal.ts

```diff
@@ -101,7 +101,7 @@ export class CircuitInternal {
             case "SetComponentPortsOp": {
                 const [newPorts, oldPorts] = op.inverted ? [op.oldPorts, op.newPorts] : [op.newPorts, op.oldPorts];
 
-                if (newPorts && !IsValidPortList(this.getComponentInfoByID(op.component), newPorts))
+                if (newPorts && newPorts.length != 0 && !IsValidPortList(this.getComponentInfoByID(op.component), newPorts))
```

Comment:
Use `!==`

Reply:
this was my workaround for when i setting ports to {} would cause issues. I just put it back to how it originally was again.